tests.py

```diff
@@ -246,12 +247,64 @@ def test_deferred_close_by_archive(self):
         z.close()
 
 
+import base64
+
+# ZIP_CONTENT is base64 encoded password protected zip file with password: 'pwd' and following contents:
+# unzip -l /tmp/zzz.zip 
+#Archive:  /tmp/zzz.zip
+#  Length      Date    Time    Name
+#---------  ---------- -----   ----
+#        9  08-09-2022 19:29   test.txt
+#---------                     -------
+#        9                     1 file
+
+ZIP_CONTENT='UEsDBAoACQAAAKubCVVjZ7b1FQAAAAkAAAAIABwAdGVzdC50eHRVVAkAA5K18mKStfJid' + \
+        'XgLAAEEAAAAAAQAAAAA5ryoP1rrRK5apjO41YMAPjpkWdU3UEsHCGNntvUVAAAACQAAAF' + \
+        'BLAQIeAwoACQAAAKubCVVjZ7b1FQAAAAkAAAAIABgAAAAAAAEAAACkgQAAAAB0ZXN0LnR' + \
+        '4dFVUBQADkrXyYnV4CwABBAAAAAAEAAAAAFBLBQYAAAAAAQABAE4AAABnAAAAAAA='
+
+ITEM_CONTENT='test.txt\n'
+ITEM_NAME='test.txt'
+
+
+class TestPasswordProtection(unittest.TestCase):
```

Comment:
Can you add a test for creating an archive with a password?

Reply:
Done